File: src/commonMain/kotlin/com/xebia/functional/textsplitters/BaseTextSplitter.kt
package com.xebia.functional.textsplitters

import com.xebia.functional.domain.Document

interface BaseTextSplitter {
    suspend fun splitText(text: String): List<String>
    suspend fun splitDocuments(documents: List<Document>): List<Document>
    suspend fun splitTextInDocuments(text: String): List<Document>
}

File: scala/src/main/scala/com/xebia/functional/textsplitters/BaseTextSplitter.scala
package com.xebia.functional.scala.textsplitters

import com.xebia.functional.scala.domain.Document

trait BaseTextSplitter[F[_]]:
  def splitDocuments(documents: List[Document]): F[List[Document]]
  def splitText(text: String): F[List[String]]
  def splitTextInDocuments(text: String): F[List[Document]]
